Server/Server.Core/Server.Core.Static/Server.hpp

```diff
@@ -20,230 +20,120 @@ class PostgreRepositoryManager;
-     * @param const std::shared_ptr<Connection>& client - for connect messages to the client or server.
-     * @param Message& message - for use the message header functionality.
-     */
-    void messageStoreRequest(const std::shared_ptr<Connection>& client, Message& message) const;
-
-    /**
-     * @brief The method for checking answer history extraction.
-     * @param const std::shared_ptr<Connection>& client - for connect messages to the client or server.
-     * @param Message& message - for use the message header functionality.
-     */
-    void replyHistoryRequest(const std::shared_ptr<Connection>& client, Message& message) const;
-
-    /**
-     * @brief The method for checking repeatly the retrieval of stored messages.
-     * @param const std::shared_ptr<Connection>& client - for connect messages to the client or server.
-     * @param Message& message - for use the message header functionality.
-     */
-    void replyStoreRequest(const std::shared_ptr<Connection>& client, Message& message) const;
-
-    /**
-     * @brief The method for checking of deletion a message request.
-     * @param const std::shared_ptr<Connection>& client - for connect messages to the client or server.
-     * @param Message& message - for use the message header functionality.
-     */
-    void messageDeleteRequest(const std::shared_ptr<Connection>& client, Message& message) const;
-
-    /**
-     * @brief The method for checking the editing of a query.
-     * @param const std::shared_ptr<Connection>& client - for connect messages to the client or server.
-     * @param Message& message - for use the message header functionality.
+     * @brief Method for connecting to a client, works asynchronously.
      */
-    void messageEditRequest(const std::shared_ptr<Connection>& client, Message& message) const;
+    void waitForClientConnection();
 
     /**
-     * @brief The method for checking the reaction.
-     * @param const std::shared_ptr<Connection>& client - for connect messages to the client or server.
-     * @param Message& message - for use the message header functionality.
+     * @brief Method for accepting to a client, works asynchronously.
      */
-    void messageReactionRequest(const std::shared_ptr<Connection>& client, Message& message) const;
+    void acceptingClientConnection(const std::error_code& error, asio::ip::tcp::socket& socket);
 
     /**
-     * @brief The method for checking the registration request.
-     * @param const std::shared_ptr<Connection>& client - for connect messages to the client or server.
-     * @param Message& message - for use the message header functionality.
+     * @brief Default constructor.
+     * @details Initialize Server object. After you steel need to initialize network and database connection.
      */
-    void registrationRequest(const std::shared_ptr<Connection>& client, Message& message) const;
+    Server() = default;
```

Comment:
Shouldn't the server have a different builder without taking the builder server into account?
And if you need to run a server without a builder?

Reply:
I think that's a build pattern and from this point every server instance must be build by builder and that means that we maybe(not so sure) can't run server without builder

Reply:
Two builders for 1 instance, quite strange
I don't see any advantages from this

Reply:
> Two builders for 1 instance, quite strange I don't see any advantages from this

Which two instances are we talking about?